src/rgw/rgw_perf_counters.cc

```diff
@@ -3,69 +3,158 @@
+  pcb->add_u64_counter(l_rgw_pubsub_push_failed, "pubsub_push_failed", "Pubsub events failed to be pushed to an endpoint");
+  pcb->add_u64(l_rgw_pubsub_push_pending, "pubsub_push_pending", "Pubsub events pending reply from endpoint");
+  pcb->add_u64_counter(l_rgw_pubsub_missing_conf, "pubsub_missing_conf", "Pubsub events could not be handled because of missing configuration");
   
-  perfcounter = plb.create_perf_counters();
-  cct->get_perfcounters_collection()->add(perfcounter);
+  pcb->add_u64_counter(l_rgw_lua_script_ok, "lua_script_ok", "Successfull executions of lua scripts");
+  pcb->add_u64_counter(l_rgw_lua_script_fail, "lua_script_fail", "Failed executions of lua scripts");
+  pcb->add_u64(l_rgw_lua_current_vms, "lua_current_vms", "Number of Lua VMs currently being executed");
+}
+
+static void add_rgw_op_counters(PerfCountersBuilder *lpcb) {
+  // description must match general rgw counters description above
+  lpcb->set_prio_default(PerfCountersBuilder::PRIO_USEFUL);
+
+  lpcb->add_u64_counter(l_rgw_op_put, "put_ops", "Puts");
+  lpcb->add_u64_counter(l_rgw_op_put_b, "put_b", "Size of puts");
+  lpcb->add_time_avg(l_rgw_op_put_lat, "put_initial_lat", "Put latency");
+
+  lpcb->add_u64_counter(l_rgw_op_get, "get_ops", "Gets");
+  lpcb->add_u64_counter(l_rgw_op_get_b, "get_b", "Size of gets");
+  lpcb->add_time_avg(l_rgw_op_get_lat, "get_initial_lat", "Get latency");
+
+  lpcb->add_u64_counter(l_rgw_op_del_obj, "del_obj_ops", "Delete objects");
+  lpcb->add_u64_counter(l_rgw_op_del_obj_b, "del_obj_bytes", "Size of delete objects");
+  lpcb->add_time_avg(l_rgw_op_del_obj_lat, "del_obj_lat", "Delete object latency");
+
+  lpcb->add_u64_counter(l_rgw_op_del_bucket, "del_bucket_ops", "Delete Buckets");
+  lpcb->add_time_avg(l_rgw_op_del_bucket_lat, "del_bucket_lat", "Delete bucket latency");
+
+  lpcb->add_u64_counter(l_rgw_op_copy_obj, "copy_obj_ops", "Copy objects");
+  lpcb->add_u64_counter(l_rgw_op_copy_obj_b, "copy_obj_bytes", "Size of copy objects");
+  lpcb->add_time_avg(l_rgw_op_copy_obj_lat, "copy_obj_lat", "Copy object latency");
+
+  lpcb->add_u64_counter(l_rgw_op_list_obj, "list_obj_ops", "List objects");
+  lpcb->add_time_avg(l_rgw_op_list_obj_lat, "list_obj_lat", "List objects latency");
+
+  lpcb->add_u64_counter(l_rgw_op_list_buckets, "list_buckets_ops", "List buckets");
+  lpcb->add_time_avg(l_rgw_op_list_buckets_lat, "list_buckets_lat", "List buckets latency");
+}
+
+std::shared_ptr<PerfCounters> create_rgw_counters(const std::string& name, CephContext *cct) {
+  std::string_view key = ceph::perf_counters::key_name(name);
+  if (rgw_op_counters_key.compare(key) == 0) {
+    PerfCountersBuilder pcb(cct, name, l_rgw_op_first, l_rgw_op_last);
+    add_rgw_op_counters(&pcb);
+    std::shared_ptr<PerfCounters> new_counters(pcb.create_perf_counters());
+    cct->get_perfcounters_collection()->add(new_counters.get());
+    return new_counters;
+  } else {
```

Comment:
@cbodley I wasn't sure what to have as the default behavior for this function. We always want to return a valid PerfCounters* right? As of now the only types of counters are the op_counters and the other frontend counters

Reply:
`rgw_op_counters_key` are the only counters stored in the cache, right? this should probably `ceph_assert()` on any other string

if we add caches for other kinds of counters, they should use a different factory function